Result/Result.swift

```diff
@@ -73,6 +73,18 @@ public enum Result<T, Error>: Printable, DebugPrintable {
 			ifSuccess: transform,
 			ifFailure: Result<U, Error>.failure)
 	}
+    
+    /// Returns `self.value` if this result is a .Success, or the given value otherwise. Equivalent with `??`
+    public func recover(value: T) -> T {
+        return self.value ?? value
+    }
+    
+    /// Returns this result if it is a .Success, or the given result otherwise. Equivalent with `??`
+    public func recoverWith(result: Result<T,Error>) -> Result<T,Error> {
```

Comment:
AIUI this one is not equivalent with `??`, because `??` doesn’t take a function on the right-hand side.